### DISCUSSION TITLE
blocking_invoke_from_main_loop not allowing to return void?

### DISCUSSION THREAD

**Discussion by DaMilyutin** [slint-ui/slint#619] [Q&A] [answered]
Have big list of error messages when [using C++]

```
            sixtyfps::blocking_invoke_from_event_loop([&](){ 
                //return 0; // uncomment: trick to make it compile
            });
```

[build-log.txt](https://github.com/sixtyfpsui/sixtyfps/files/7437833/build-log.txt)

Everithing, however, compiles when
```
            sixtyfps::invoke_from_event_loop([&](){ 
                //return 0; // no need for trick to make it compile
            });
```
Am I doing it wrong or is it a bug?

**Comment by tronical** [comment]
I think that it is a bug that closures returning void can't be used with `sixtyfps::blocking_invoke_from_event_loop`. I'll create a ticket for this. Your workaround is perfectly valid :-)

**Comment by ogoffart** [comment, marked as answer]
Note that this is now fixed in master ( see issue #623 )
